straxen/plugins/led_calibration.py

```diff
@@ -141,3 +142,80 @@ def get_area(records, led_window):
     Area['area']    = Area['area']/float(len(end_pos))
         
     return Area
+
+
+@export
+@strax.takes_config(
+    strax.Option('channel_map', track=False, type=immutabledict,
+                 help="immutabledict mapping subdetector to (min, max) "
+                      "channel number."),
+)
+class nVetoExtTimings(strax.Plugin):
+    """
+    Plugin which computes the time differnce from external trigger timing to hitlets_nv.
+    The nearlest `time` of `raw_records_nv` before `hitlets_nv` time are used as the
+    external trigger timings.
+
+    Note:
+        Channel dependence for trigger timing has been ignored. It should be the same,
+        But unfortunetely not.
+    """
+    depends_on = ('raw_records_nv', 'hitlets_nv')
+    provides = 'ext_timings_nv'
+    data_kind = 'hitlets_nv'
+
+    compressor = 'zstd'
+    __version__ = '0.0.1'
+
+    def infer_dtype(self):
+        dtype = []
+        dtype += strax.time_dt_fields
+        dtype += [(('Delta time from trigger timing [ns]', 'delta_time'), np.int16),]
+        return dtype
+
+    def setup(self):
+        self.nv_pmt_start = self.config['channel_map']['nveto'][0]
+        self.nv_pmt_stop = self.config['channel_map']['nveto'][1] + 1
+
+
+    def compute(self, hitlets_nv, raw_records_nv):
+
+        rr_nv = raw_records_nv[raw_records_nv['record_i']==0]
+        pulses = np.zeros(len(rr_nv), dtype=self.pulse_dtype())
+        pulses['time'] = rr_nv['time']
+        pulses['endtime'] = rr_nv['time'] + \
+                                    rr_nv['pulse_length'] * rr_nv['dt']
+        pulses['channel'] = rr_nv['channel']
+
+        ext_timings_nv = np.zeros_like(hitlets_nv, dtype=self.infer_dtype())
```

Comment:
```suggestion
        ext_timings_nv = np.zeros_like(hitlets_nv, dtype=self.dtype)
```
infer_dtype will store your dtype as a property of the plugin hence you can simply call `self.dtype`